src/Math/babylon.math.ts

```diff
@@ -2034,10 +2045,7 @@ module BABYLON {
          * @returns the current updated Vector3
          */
         public copyFrom(source: Vector3): Vector3 {
-            this.x = source.x;
-            this.y = source.y;
-            this.z = source.z;
-            return this;
+            return this.copyFromFloats(source.x, source.y, source.z);

```

Comment:
not sure this is faster

Reply:
Neither am I. Actually from my little javascript experience, such low level performance are very hard to predict as there are very engine dependant. On one hand, one can think that it would be faster as we avoid one function call. On the other hand, the Jitter is likely to inline the function call anyway and always going through the same function would make it considered "hot" by the engine optimizer so it is more likely to be highly optimized...

So I tend to share as much code as possible, as if it it does not really affect performance, it reduces the size of the code and avoiding code duplication tends to to makes thing easier to understand, test and maintain.

But I can revert it if you prefer, it is more "your" code base than mine, it is up to you :)